### ISSUE TITLE
Versioning libhl.so file on Linux

### ISSUE THREAD

**Issue by l0go** [HaxeFoundation/hashlink#698]
Hi,
I am currently working towards packaging HashLink for the Fedora Linux distribution. One of their guidelines is that .so libraries are versioned, preferably by upstream (you!). This means that ``libhl.so`` should be something like ``libhl.so.0`` with 0 incrementing on ABI changes. Would it be possible for this to happen?

**Comment by andyli** [comment]
I think when building with CMake, the library is already versioned. See usage of `SOVERSION` in `CMakeLists.txt`.

Someone have to remember update it correctly though. It's a bit trickly - the software version number does not implies SOVERSION number. What I wrote initially in `CMakeLists.txt` that `SOVERSION` = `${HL_VERSION_MAJOR}` is not quite correct. SOVERSION follows semver **strictly**, i.e. whenever there is any breaking change, it goes up.

**Comment by tobil4sk** [comment]
There are tools we can use that compare the ABI with previous versions to check if SOVERSION ought to have changed. We can set up github actions that fail if the ABI changes but the so version remains the same.

MbedTLS does this, and I know they are very diligent with changing the so version on any ABI change :sweat_smile: https://github.com/Mbed-TLS/mbedtls/blob/4b9141ccf45e92d42152f9ec19cc69fab31b066b/scripts/abi_check.py

**Comment by andyli** [comment]
Yes, I think it's a good direction. Debian's [packaging system](https://www.debian.org/doc/manuals/maint-guide/advanced.en.html#librarysymbols) is able to do this as well.

Not sure if there is a standalone tool hashlink can adopt, or have to write a custom script like MbedTLS.

**Comment by l0go** [comment]
MbedTLS utilizes these tools, which are recommended by Fedora:
https://github.com/lvc/abi-dumper/
https://github.com/lvc/abi-compliance-checker
